### ISSUE TITLE
Map fields disappearing after save

### ISSUE THREAD

**Issue by ljclark** [ethercreative/simplemap#160]
### Description

The Map and it's associated fields (search, address etc) are present on a new entry. When you save the entry, it's saving the location and rendering the map on the front end as expected. 

However, if you try and edit that map data in the entry again, none of the inputs for the map field are visible. 

There's this error in the console when you try edit the entry which I'm assuming is related to this. All other fields are fine, its just the maps field that's breaking. 

```
vendor.js:13 TypeError: Cannot read property '$on' of undefined
    at p.mounted (sidebar.js:9)
    at We (vendor.js:13)
    at Object.insert (vendor.js:13)
    at y (vendor.js:13)
    at Mo.__patch__ (vendor.js:13)
    at Mo.t._update (vendor.js:13)
    at Mo.n (vendor.js:13)
    at Xe.get (vendor.js:13)
    at new Xe (vendor.js:13)
    at vendor.js:13
```
### Steps to reproduce

1. Create an entry using the map field
2. Save the entry
3. Go back and edit the entry, and the Map fields won't be visible. 

### Additional info

- Craft version: Craft Pro 3.1.25
- Maps version: 3.4.11
- PHP version: 7.1.20
- Database driver & version: MySQL 5.5.62
- Other Plugins - see screenshot below:

<img width="547" alt="Screenshot 2019-05-08 at 21 33 37" src="https://user-images.githubusercontent.com/8060816/57406692-54705280-71d9-11e9-8dfb-541111b6c29c.png">

**Comment by codeartaus** [comment]
I'm also getting this issue, as well as a 404 for the file /cpresources/2ed3d3ff/js/apifetch.js 

Did you find a fix for this one?

**Comment by codeartaus** [comment]
I just found a fix for mine, it was another plugin causing a js error, that in turn was breaking the maps

**Comment by jacobgraf** [comment]
I am having this same issue and it's definitely a frustrating one. Is there a fix on the way?

**Comment by Tam** [comment]
@jacobgraf Do you share any of the plugins @ljclark listed? The likely cause seems to be that another plugin is interfering with Maps' JS.

**Comment by jacobgraf** [comment]
Only Redactor is common between us.

**Comment by elfacht** [comment]
@codeartaus Would you mind sharen the plugin with us which was causing the error?

**Comment by elfacht** [comment]
I'm getting this error:

`You have included the Google Maps JavaScript API multiple times on this page. This may cause unexpected errors.`

Saving an entry without entering a location into the maps field will break the output. So I have to re-enter the location on every save. Frustrating. Especially in combination with this issue: https://github.com/ethercreative/simplemap/issues/165#issuecomment-504755775

**Edit:**

The JS error is caused because of multiple includes in [MapField.php](https://github.com/ethercreative/simplemap/blob/v3/src/fields/MapField.php#L547-L579)

```php 
	// Map Services
// ---------------------------------------------------------------------

if (strpos($settings->mapTiles, 'google') !== false)
{
  $view->registerJsFile(
    'https://maps.googleapis.com/maps/api/js?libraries=places&key=' .
    $settings->mapToken
  );
}
elseif (strpos($settings->mapTiles, 'mapkit') !== false)
{
  $view->registerJsFile(
    'https://cdn.apple-mapkit.com/mk/5.x.x/mapkit.js'
  );
}

// Geo Services
// ---------------------------------------------------------------------

if ($settings->geoService === GeoEnum::GoogleMaps)
{
  $view->registerJsFile(
    'https://maps.googleapis.com/maps/api/js?libraries=places&key=' .
    $settings->geoToken
  );
}
elseif ($settings->geoService === GeoEnum::AppleMapKit)
{
  $view->registerJsFile(
    'https://cdn.apple-mapkit.com/mk/5.x.x/mapkit.js'
  );
}
```

Removing either will still work and eleminates the error, but the bug with the disappearing data will still occur.

And this seems to be why Mapbox doesn't work: it will be not included. I tried Mapbox with my API, but with no results.

Output with correct saving:

```json
{
   "googlePlace":{
      "administrative_area_level_1":"Hamburg",
      "administrative_area_level_2":"",
      "administrative_area_level_3":"",
      "administrative_area_level_4":"",
      "administrative_area_level_5":"",
      "airport":"",
      "bus_station":"",
      "colloquial_area":"",
      "establishment":"",
      "floor":"",
      "intersection":"",
      "locality":"Hamburg",
      "natural_feature":"",
      "neighborhood":"",
      "park":"",
      "parking":"",
      "point_of_interest":"",
      "political":"",
      "post_box":"",
      "postal_code":"20259",
      "postal_code_prefix":"",
      "postal_town":"",
      "premise":"",
      "room":"",
      "route":"Osterstraße",
      "street_address":"",
      "street_number":"98",
      "sublocality":"",
      "sublocality_level_1":"Bezirk Eimsbüttel",
      "sublocality_level_2":"",
      "sublocality_level_3":"",
      "sublocality_level_4":"",
      "sublocality_level_5":"",
      "subpremise":"",
      "train_station":"",
      "transit_station":"",
      "administrative_area_level_1_short":"",
      "administrative_area_level_2_short":"",
      "administrative_area_level_3_short":"",
      "administrative_area_level_4_short":"",
      "administrative_area_level_5_short":"",
      "airport_short":"",
      "bus_station_short":"",
      "colloquial_area_short":"",
      "establishment_short":"",
      "floor_short":"",
      "intersection_short":"",
      "locality_short":"",
      "natural_feature_short":"",
      "neighborhood_short":"",
      "park_short":"",
      "parking_short":"",
      "point_of_interest_short":"",
      "political_short":"",
      "post_box_short":"",
      "postal_code_short":"",
      "postal_code_prefix_short":"",
      "postal_town_short":"",
      "premise_short":"",
      "room_short":"",
      "route_short":"",
      "street_address_short":"",
      "street_number_short":"",
      "sublocality_short":"",
      "sublocality_level_1_short":"",
      "sublocality_level_2_short":"",
      "sublocality_level_3_short":"",
      "sublocality_level_4_short":"",
      "sublocality_level_5_short":"",
      "subpremise_short":"",
      "train_station_short":"",
      "transit_station_short":"",
      "number":"98",
      "address":"Osterstraße, Bezirk Eimsbüttel",
      "city":"Hamburg",
      "postcode":"20259",
      "county":"",
      "state":"Hamburg",
      "country":"Deutschland"
   }
}
```

After re-saving:

```json
{
   "googlePlace":{
      "number":"98",
      "address":"Osterstraße, Bezirk Eimsbüttel",
      "city":"Hamburg",
      "postcode":"20259",
      "county":"",
      "state":"Hamburg",
      "country":"Deutschland"
   }
}
```

**Comment by elfacht** [comment]
For me this bug disappears after updating Craft to 3.2

**Comment by jacobgraf** [comment]
Works fine in 3.2 for me too! 🎉

**Comment by elfacht** [comment]
Aaaaand it happens again after updating to 3.6.0

**Comment by ljclark** [comment]
Upgrading to C3.2 and Maps 3.6 has made Maps complete unusable for me now. I don't even get the maps field or input on a brand new entry! Any update on this?

**Comment by dbig** [comment]
Similar for me - a new install of SimpleMap 3.6 into C3.2.4.1 results in entered address not searching/saving.  Empty address fields when entry opened after saving.

**Comment by Tam** [comment]
This should be fixed in 3.6.1.
